src/parsers/rules.jl

```diff
@@ -27,6 +27,7 @@ enable!(p::AbstractParser, fn, rules::Union{Tuple,Vector}) = (foreach(r -> enabl
 enable!(p::AbstractParser, fn, rule) = enable!(p, fn, fn(rule))
 
 function enable!(p::AbstractParser, rule)
+    rule ∈ p.rules && return p
```

Comment:
I am not sure if this is actually the best behavior here. I wonder if it should instead either throw an error or replace an existing rule, should one already exist (thinking of the use case where you might want to override a rule).

Reply:
Probably throw an error, we could add an `override!` that is meant for replacing existing rules and doesn't throw, that way it's a bit more explicit that that is the users' intension.